### DISCUSSION TITLE
Uploading, Storing, and Transferring MP3, PDF, and XLS Files: Data Integrity Concerns

### DISCUSSION THREAD

**Discussion by Upsite-cor** [react-dropzone/react-dropzone#1359] [Q&A]
I am currently working on a project that involves users dragging and dropping MP3, PDF, and XLS files for upload. I have a few questions regarding file handling after the upload process:

- **File Integrity**: Once these files are uploaded into a variable or an array, is all the data preserved intact?

- **Database Storage**: Is it feasible to store these uploaded files in a database while ensuring their integrity?

- **Cross-Server Utilization**: If these files are stored in a database, can they later be extracted and moved to a different server for use? Specifically, I am interested in whether an MP3 file, for example, can be uploaded, stored in a database, and then retrieved to be used on another server without data loss.

I would appreciate any insights on this :)

**Comment by rolandjitsu** [comment]
@Upsite-cor this package does not handle file uploads, it simply provides an interface for file selection and drag'n'drop of files. The data reading part and storage/upload to server is handled by the user (generally using https://developer.mozilla.org/en-US/docs/Web/API/FileReader - https://react-dropzone.js.org/#usage).

**Reply by rolandjitsu** [reply]
As for data integrity, you can expect that the data you read using the https://developer.mozilla.org/en-US/docs/Web/API/FileReader API to have no integrity issues (as long as there are no bugs in the browser APIs).
